packages/common/src/api/tan-query/jupiter/utils.ts

```diff
@@ -0,0 +1,174 @@
+import { AudiusSdk } from '@audius/sdk'
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  createAssociatedTokenAccountIdempotentInstruction,
+  createCloseAccountInstruction,
+  createTransferCheckedInstruction,
+  getAccount,
+  getAssociatedTokenAddressSync
+} from '@solana/spl-token'
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+
+import { JUPITER_PROGRAM_ID } from './constants'
+import { UserBankManagedTokenInfo } from './types'
+
+export async function addUserBankToAtaInstructions({
+  tokenInfo,
+  userPublicKey,
+  ethAddress,
+  amountLamports,
+  sdk,
+  feePayer,
+  instructions
+}: {
+  tokenInfo: UserBankManagedTokenInfo
+  userPublicKey: PublicKey
+  ethAddress: string
+  amountLamports: bigint
+  sdk: any
+  feePayer: PublicKey
+  instructions: TransactionInstruction[]
+}): Promise<PublicKey> {
+  const mint = new PublicKey(tokenInfo.mintAddress)
+  const ata = getAssociatedTokenAddressSync(mint, userPublicKey, true)
+
+  try {
+    await getAccount(sdk.services.solanaClient.connection, ata)
+  } catch (e) {
+    instructions.push(
+      createAssociatedTokenAccountIdempotentInstruction(
```

Comment:
nit: since this is idempotent, we can add it every time (if it fits!)

i like that we're only doing it conditionally given our size constraint, but I think if we're closing it every time then it'll end up getting added every time anyway...